### PR TITLE
Add a GitHub action file to run neptune auto-pr

### PR DESCRIPTION
**Pull request by ibuclaw**
~Currently just a skeleton 'hello world' program, will be filled in with actual job later.~

Should just work, though can't test here in the PR as github doesn't allow passing secrets to actions from pull requests.

### CODE REVIEW

**Comment by llucax** [comment]
Travis says no.

**Comment by ibuclaw** [comment]
> Travis says no.

Travis said no to the last pull request here as well, so I wonder if there's something else broken, or some dependencies have drifted away and are now incompatible.

**Comment by ibuclaw** [comment]
> > Travis says no.
> 
> Travis said no to the last pull request here as well, so I wonder if there's something else broken, or some dependencies have drifted away and are now incompatible.

```
dependency "vibe-d:data" version="~>0.8"
```
Latest version is 0.9.0, so I'll update the dependency to restrict it to just 0.8.x series.

File: .github/workflows/autopr.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - name: Setup dlang environment
+      uses: dlang-community/setup-dlang@v1
+      with:
+        compiler: dmd-2.084.1
+
+    # Checks-out the neptune repository under $GITHUB_WORKSPACE
+    - name: Checkout sociomantic-tsunami/neptune
+      uses: actions/checkout@v2
+      with:
+        submodules: 'recursive'
```
Comment (Geod24) [inline .github/workflows/autopr.yml]: Why?

**Comment by Geod24** [comment]
@ibuclaw : How could this be related to dlang-community when Travis is failing ? :|

**Comment by ibuclaw** [comment]
@Geod24 travis is failing for other reasons that I'm trying to determine in #259.

You've missed the part when I was [testing actions](https://github.com/sociomantic-tsunami/neptune/actions).  It won't run against PRs now as I've set it up to be a cron.

Here's a [failing one](https://github.com/sociomantic-tsunami/neptune/runs/999029846?check_suite_focus=true), and this one is with [debugging on](https://github.com/sociomantic-tsunami/neptune/runs/999045999?check_suite_focus=true).  You can see messages like:
```
Package internal not found at registry at https://code.dlang.org/ (404): HTTP request returned status code 404 (Not Found)
No versions for internal for registry at https://code.dlang.org/ (fallback ["registry at http://code.dlang.org/", "registry at https://code-mirror.dlang.io/", "registry at https://code-mirror2.dlang.io/", "registry at https://dub-registry.herokuapp.com/"])
```
Which is a symptom of the curl bug.

File: .github/workflows/autopr.yml (hunk 1 above)
Comment (ibuclaw) [reply .github/workflows/autopr.yml]: I used the old Jenkinsfile as a point of reference, can improve upon it though.

**Comment by Geod24** [comment]
Well https://code.dlang.org/packages/internal

**Comment by Geod24** [comment]
Yeah it's trying to look up https://github.com/sociomantic-tsunami/neptune/blob/v0.x.x/src/internal/dub.sdl on the registry, no wonder it gets a 404

**Comment by ibuclaw** [comment]
Travis failure has been fixed in #260.

**Comment by ibuclaw** [comment]
This is ready.

File: .github/workflows/autopr.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+
+    # Setting up environment with a D compiler
+    - name: Setup dlang environment
+      uses: dlang-community/setup-dlang@v1
+      with:
+        compiler: dmd-2.084.1
```
Comment (Geod24) [inline .github/workflows/autopr.yml:18]: Why such an old compiler ?

File: .github/workflows/autopr.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    # Checks-out the neptune repository under $GITHUB_WORKSPACE
+    - name: Checkout sociomantic-tsunami/neptune
+      uses: actions/checkout@v2
+      with:
+        submodules: 'true'
```
Comment (Geod24) [inline .github/workflows/autopr.yml]:
```suggestion
        submodules: true
```

File: .github/workflows/autopr.yml (hunk 2 above)
Comment (ibuclaw) [reply .github/workflows/autopr.yml:18]: We need to fix deprecation errors in neptune first.  Travis is still using 2.080/81.*.  This should be dealt with in another pull though.

**Review by Geod24 — APPROVED** [review]

**Comment by ibuclaw** [comment]
I'll merge this, as the next scheduled run is coming up tomorrow, and it allows removing a piece of the old Jenkins infrastructure.

**Pull request merged**